text/0000-promote-riscv64gc-unknown-linux-gnu-to-tier1-without-host-tools.md

```diff
@@ -0,0 +1,185 @@
+
+It is fair to say that the target specific ecosystem has been a viable target for some time now, and that this is likely to continue into the long term.
+
+> 2.b. The long term viability of supporting the target should be clear.
+
+It is hard to concretely quantify this aspect. This work was initiated and supported by the [RISE Project](https://riseproject.dev/) and there is an intention to continue to support the target and eventually propose a *Tier 1 with Host Tools* RFC when sufficiently fast hardware exists.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+Adopting the platform would require additional commitments by the Rust project. Future contributions may impact the target and cause changes to become delayed or halted entirely due to problems on the target.
+
+In general, it should be uncomplicated for contributors to build for and use a `riscv64gc-unknown-linux-gnu` emulator like `qemu`, `docker`, or `lima`. Additionally, the platform is a `*-unknown-linux-gnu` target which is generally quite well understood, contributors do not need to learn what could be an otherwise unfamiliar operating system.
+
+This target does not place significant burdens on the project that would not be present on any other target.
+
+# Rationale and alternatives
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+There exist two alternatives: Promoting the target to Tier 1 (with host tools), or not promoting the target at all.
+
+## Tier 1 (with host tools)
+
+The [Tier 1 Target Tier Policy](https://doc.rust-lang.org/nightly/rustc/target-tier-policy.html#tier-1-target-policy) section 1.e states:
+
+> 1.e. Building the target and running the testsuite for the target must not take
+>   substantially longer than other targets...
+
+During testing on [Scaleway Elastic Metal RV1](https://labs.scaleway.com/en/em-rv1/) it was determined that running a full `x.py test` run takes roughly 6 hours. A similar amount of time was taken on a 6 CPU, 16GB RAM VM.
+
+During this testing, it was noted that all tests pass.
+
+It's fair to conclude that existing available virtualization and hardware for `riscv64gc-unknown-linux-gnu` takes substantially longer to run the test suite than other targets. If sufficiently fast hardware existed, this RFC would be for Tier-1 with host tools instead.
+
+## Not promoting the target
+
+Not promoting the target could lead to a situation where the `riscv64gc-unknown-linux-gnu` tests are no longer passing, and this could impact users.
+
+Anecdotally, not having the Tier 1 'badge' has been seen to become an obstacle to increasing mindshare in Rust for this target. Organisations tend to associate a Tier 1 categorisation with better quality, suitability for key projects, longevity etc. With a reasonably justified Tier 1 'badge' in place, the likelihood is that such organisations will tend to pick up and promote the use of Rust in production.
```

Comment:
This should also discuss the option of having a vendor other than the rust-lang org provide the equivalent of Tier 1 support, e.g. by running CI externally and communicating equivalence properly.

That way even host tools could be tested without the infra burden of maintaining custom github runners.

Reply:
That roughly happens with additional arches on Linux distros already.